__tests__/installer.test.ts

```diff
@@ -1,298 +1,402 @@
-import * as io from '@actions/io';
-import * as os from 'os';
-import fs from 'fs';
-import path from 'path';
 import each from 'jest-each';
-import * as hc from '@actions/http-client';
+import semver from 'semver';
+import * as exec from '@actions/exec';
+import * as core from '@actions/core';
+import * as io from '@actions/io';
 import * as installer from '../src/installer';
-import {QualityOptions} from '../src/setup-dotnet';
 
 import {IS_WINDOWS} from '../src/utils';
-import {IS_LINUX} from '../src/utils';
-
-let toolDir: string;
-
-if (IS_WINDOWS) {
-  toolDir = path.join(process.env['PROGRAMFILES'] + '', 'dotnet');
-} else if (IS_LINUX) {
-  toolDir = '/usr/share/dotnet';
-} else {
-  toolDir = path.join(process.env['HOME'] + '', '.dotnet');
-}
-const tempDir = path.join(__dirname, 'runner', 'temp');
-
-process.env['RUNNER_TOOL_CACHE'] = toolDir;
-process.env['RUNNER_TEMP'] = tempDir;
-
-describe('DotnetCoreInstaller tests', () => {
-  beforeAll(async () => {
-    process.env.RUNNER_TOOL_CACHE = toolDir;
-    process.env.DOTNET_INSTALL_DIR = toolDir;
-    process.env.RUNNER_TEMP = tempDir;
-    process.env.DOTNET_ROOT = '';
-    try {
-      await io.rmRF(`${toolDir}/*`);
-      await io.rmRF(`${tempDir}/*`);
-    } catch (err) {
-      console.log(
-        `Failed to remove test directories, check the error message:${os.EOL}`,
-        err.message
+import {QualityOptions} from '../src/setup-dotnet';
+
+describe('installer tests', () => {
+  const env = process.env;
+
+  beforeEach(() => {
+    jest.resetModules();
+    process.env = {...env};
+  });
+
+  afterEach(() => {
+    process.env = env;
+  });
+
+  describe('DotnetCoreInstaller tests', () => {
+    const getExecOutputSpy = jest.spyOn(exec, 'getExecOutput');
+    const warningSpy = jest.spyOn(core, 'warning');
+    const whichSpy = jest.spyOn(io, 'which');
+    const maxSatisfyingSpy = jest.spyOn(semver, 'maxSatisfying');
+
+    describe('installDotnet() tests', () => {
+      whichSpy.mockImplementation(() => Promise.resolve('PathToShell'));
+
+      it('should throw the error in case of non-zero exit code of the installation script. The error message should contain logs.', async () => {
+        const inputVersion = '3.1.100';
+        const inputQuality = '' as QualityOptions;
+        const errorMessage = 'Fictious error message!';
```

Comment:
```suggestion
        const errorMessage = 'Fictitious error message!';
```

Reply:
Updated this typo everywhere in the tests.